lib/route.js

```diff
@@ -99,6 +99,10 @@ function buildRouting (options) {
 
   // Convert shorthand to extended route declaration
   function prepareRoute (method, url, options, handler) {
+    if (url && typeof url !== 'string') {
+      throw new Error(`URL must be a string. Received ${typeof url}`)
```

Comment:
Could you please use our error codes (https://github.com/fastify/fastify/blob/main/lib/errors.js)? Thanks!

Reply:
Thanks @mcollina. Added a new Error code and used the same here. I could have used Bad URL but the error message in it said invalid url component. Since our error here is specifically related to non-string URL i didn't reuse that